### ISSUE TITLE
env_or_file mechanism fails when the env is too long to be a file

### ISSUE THREAD

**Issue by TripleShahar** [microsoft/autogen#3931]
In `config_list_from_json` under `autogen/oai/openai_utils.py` in the `config_list_from_json` function, the mechanism of setting the `env_or_file` variable to be either a filename to a OSI file or the string that has the content of said file fails if the string is long enough. So in the following code:
```
# The environment variable exists. We should use information from it.
if os.path.exists(env_str):
    # It is a file location, and we need to load the json from the file.
    with open(env_str, "r") as file:
        json_str = file.read()
else:
    # Else, it should be a JSON string by itself.
    json_str = env_str
config_list = json.loads(json_str)
```
The else is never reached because `os.path.exists` raises an exception of File name too long. For example this is the `env_or_file` i've used: 

[Errno 63] File name too long. 
'{"config_list": [{"model": "gpt-4", "api_key": "<Redacted>", "tags": ["gpt-4", "tool"]}, {"model": "gpt-4-vision-preview", "api_key": "<Redacted>", "tags": ["vision"]}, {"model": "gpt-4o", "api_key": "<Redacted>",  "tags": ["gpt-4o"]}]}'

a simple check of file name length should work: 
```
if len(env_str) <= 255 and os.path.exists(env_str):
    # It is a file location, and we need to load the json from the file.
    with open(env_str, "r") as file:
        json_str = file.read()
else:
    # Else, it should be a JSON string by itself.
    json_str = env_str
config_list = json.loads(json_str)
```

**Comment by ekzhu** [comment]
I think raising an exception on a too long filename makes more sense and portal across platforms.

**Comment by TripleShahar** [comment]
Yeah, but this isn't *only* a filename. For the case where `env_or_file` is `env` and not `file` it is expected that `env_str` could be very long if you have a lot of configurations in your config list.

**Comment by ekzhu** [comment]
Right. It is a bug. Though fixing it by skipping long filename by a fixed number may not be portable across platforms. I was suggesting checking it by catching the exception instead.
